### PR TITLE
Cleanup query prepare/execute/stream

### PR DESCRIPTION
**Pull request by fishcakez**
* Remove parameter_types and types from query struct
* Use with for flow control
* Use status flags to detect extra ok response
* Use cursor flags on execute

### CODE REVIEW

**Comment by fishcakez** [comment]
@liveforeverx sorry for the large refactor but wanted to avoid all this data we aren't using. It was hard to follow the recursive logic so I hope this is ok. Would appreciate a 0.8 so we can move forward in Ecto. Note that this commit allows ecto mysql to handle live migrations without any errors (assuming types can be cast to the new column type).

**Comment by fishcakez** [comment]
There are a few other changes I might like to do here before Ecto 2.1 actually. I would like to get improved handling between prepared and simple queries.

**Comment by liveforeverx** [comment]
Thank you! Agree, it is really easier to read.
